docs/spec.md

```diff
@@ -170,6 +174,82 @@ Informs the client that channels are no longer available.
 }
 ```
 
+### Client Advertise
+
+- Informs the server about available client channels. Note that hhe client is only allowed to advertise channels if the server previously declared that it has the `clientPublish` [capability](#server-info).
+
+#### Fields
+
+- `op`: string `"advertise"`
+- `channels`: array of:
+  - `id`: number chosen by the client. The client may not reuse ids across multiple advertised channels.
```

Comment:
We said this about channelId for server advertise:

> The server may reuse ids when channels disappear and reappear, but only if the channel keeps the exact same topic, encoding, schemaName, and schema. Clients will use this unique id to cache schema info and deserialization routines.

Should we do something like this here too? Currently it doesn't sound like the same id is allowed to be reused even if it's unadvertised in between? Keeping the restriction as you have it is fine too, but might want to make it more clear that even unadvertising does not allow you to use the same id again.

Reply:
We can allow the client to reuse channel IDs if they have been unadvertised in between. I see no problem in that as the server will use the channel ID only to lookup published channels of the client